src/resmom/catch_child.c

```diff
@@ -2694,6 +2694,36 @@ mom_set_use_all(void)
 void
 job_purge_mom(job *pjob)
 {
+	extern pbs_list_head task_list_event;
+	extern int post_run_hook(struct work_task *wtask);
+
+	struct work_task *wtask = NULL;
+	struct work_task *wtask_next = NULL;
+
+	/* Check for other task lists */
+	for (wtask = (struct work_task *) GET_NEXT(task_list_event); wtask; wtask = wtask_next) {
```

Comment:
I don't think this is right fix and place, this will kill run performance as for every job purge we will go through whole task event list. Also this function is just wrapper of actual job purge vs mock run job purge. And in mock run mode mom doesn't run any hook (aka no fork at all).

I think what we should do is hang work task pointer in pjob struct, inside run_hook and in job_purge() just make that work task's wt_param2 null, and in post_run_hook check for null of param2, if null don't do anything just return.
that way we don't have to delete task also, or if we want we can delete task also with ensuring child process is also killed and reaped.